core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -607,6 +603,74 @@ case class PSBT(
                   logger.info(
                     s"No UTXO record was provided for input $index, hence no validation was done for this input")
 
+                  Success(tx)
+              }
+            case (None, wtx: WitnessTransaction) =>
+              utxoOpt match {
+                case Some(utxo) =>
+                  val input = tx.inputs(index)
+                  val output = utxo.transactionSpent.outputs(
+                    input.previousOutput.vout.toInt)
+                  val txSigComponent = output.scriptPubKey match {
```

Comment:
this block really seems like it deserves its own method in a different file

Reply:
btw, I suggest that this method take `Policy.standardFlags` as a param rather than hard coding it (just to make the future easier)